src/Components/Endpoints/src/DependencyInjection/RazorComponentsServiceCollectionExtensions.cs

```diff
@@ -62,21 +62,24 @@ public static IRazorComponentsBuilder AddRazorComponents(this IServiceCollection
         services.TryAddScoped<INavigationInterception, UnsupportedNavigationInterception>();
         services.TryAddScoped<IScrollToLocationHash, UnsupportedScrollToLocationHash>();
         services.TryAddScoped<ComponentStatePersistenceManager>();
-        services.TryAddScoped<PersistentComponentState>(sp => sp.GetRequiredService<ComponentStatePersistenceManager>().State);
+        services.TryAddScoped(sp => sp.GetRequiredService<ComponentStatePersistenceManager>().State);
         services.TryAddScoped<IErrorBoundaryLogger, PrerenderingErrorBoundaryLogger>();
         services.TryAddEnumerable(
             ServiceDescriptor.Singleton<IPostConfigureOptions<RazorComponentsServiceOptions>, DefaultRazorComponentsServiceOptionsConfiguration>());
         services.TryAddScoped<EndpointRoutingStateProvider>();
         services.TryAddScoped<IRoutingStateProvider>(sp => sp.GetRequiredService<EndpointRoutingStateProvider>());
         services.TryAddScoped<AuthenticationStateProvider, ServerAuthenticationStateProvider>();
         services.AddSupplyValueFromQueryProvider();
+        services.AddSupplyValueFromPersistentComponentStateProvider();
         services.TryAddCascadingValue(sp => sp.GetRequiredService<EndpointHtmlRenderer>().HttpContext);
 
         services.TryAddScoped<ResourceCollectionProvider>();
 
         // Form handling
         services.AddSupplyValueFromFormProvider();
         services.TryAddScoped<AntiforgeryStateProvider, EndpointAntiforgeryStateProvider>();
+        services.TryAddScoped(sp => (EndpointAntiforgeryStateProvider)sp.GetRequiredService<AntiforgeryStateProvider>());
+        services.AddPersistentService<AntiforgeryStateProvider>(RenderMode.InteractiveAuto);
```

Comment:
Can you please explain `InteractiveAuto` here ? Thanks.

Reply:
Sure, the antiforgery token needs to be persisted to both contexts, interactive webassembly and interactive server. Interactive auto makes sure it gets persisted to both contexts. This is the original [registration](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Shared/src/DefaultAntiforgeryStateProvider.cs#L29) (Using the old imperative API)

Reply:
I'm confused about antiforgery token lifecycle in wasm. 
1) It's created on server side during SSR
2) persisted to store
3) ... i'm lost

Reply:
What kind of security guarantees the actual storage offers ? Is the data encrypted at rest ? Can this be abused in replay attack if the "cache" was stolen ?

Reply:
The request antiforgery token is persisted and passed into wasm, so that you can use the `<AntiforgeryToken />` component.

Antiforgery is made of two components:
* A request token that gets normally rendered as a hidden input field.
* A cookie token that gets stored in an http only cookie.

The `AntiforgeryStateProvider` persist the token as part of Blazor state into the HTTP response so that the interactive models have access to it during interactive rendering (and so that `<AntiforgeryToken>` can create the hidden input field with the token for forms that aren't handled interactively in interactive contexts)

Reply:
> What is the encryption key when this is stored into cloud persistent store?

This information is only ever written to the document as part as the HTML response, nothing here stores anything into any other location.

When the information is written to the response and is not meant to be consumed by webassembly, all the payloads use [data protection](https://learn.microsoft.com/en-us/aspnet/core/security/data-protection/introduction?view=aspnetcore-9.0) which implements authentication encryption to offer confidentiality and integrity of the data.